base/docker-integration.go

```diff
@@ -0,0 +1,320 @@
+		fatalError(`Unable to create bucket`, err)
+		err = manager.UpdateBucket(&bucketSettings2)
+		fatalError(`Unable to create bucket`, err)
+
+		err = manager.UpsertUser("local", "test_data_bucket", &userSettings)
+		fatalError(`Unable to create user`, err)
+		err = manager.UpsertUser("local", "test_shadowbucket", &userSettings1)
+		fatalError(`Unable to create user`, err)
+		err = manager.UpsertUser("local", "test_indexbucket", &userSettings2)
+		fatalError(`Unable to create user`, err)
+
+		err = cluster.Close()
+		if err != nil {
+			fmt.Println("Failed to close gocb cluster connection but will continue")
+		}
+
+		//Poll buckets to ensure they are all healthy and ready to work
+		req, err = http.NewRequest("GET", "http://localhost:8091/pools/default/buckets", nil)
+		fatalError(`Failed to poll bucket status`, err)
+		req.SetBasicAuth("Administrator", "password")
+		response, err = client.Do(req)
+		fatalError(`Failed to poll bucket status`, err)
+		var jsonBody []responseBody
+		err = json.NewDecoder(response.Body).Decode(&jsonBody)
+		fatalError(`Failed to poll bucket status`, err)
+
+		for !areBucketsReady(&jsonBody) {
+			fmt.Println("Buckets Not Ready Yet Will Retry In 1 Second")
+			time.Sleep(1 * time.Second)
+			req, _ = http.NewRequest("GET", "http://localhost:8091/pools/default/buckets", nil)
+			req.SetBasicAuth("Administrator", "password")
+			response, err = client.Do(req)
+			fatalError(`Failed to poll bucket status`, err)
+			err = json.NewDecoder(response.Body).Decode(&jsonBody)
+			fatalError(`Failed to poll bucket status`, err)
+		}
+
+		//Poll nodes / node to ensure they are all healthy and ready to work
+		var jsonBody2 responseBody
+		req, err = http.NewRequest("GET", "http://localhost:8091/pools/default", nil)
+		fatalError(`Failed to poll node status`, err)
+		req.SetBasicAuth("Administrator", "password")
+		response, err = client.Do(req)
+		fatalError(`Failed to poll node status`, err)
+		err = json.NewDecoder(response.Body).Decode(&jsonBody2)
+
+		for !isNodeReady(&jsonBody2) {
+			fmt.Println("Node Not Ready Yet Will Retry In 1 Second")
+			time.Sleep(1 * time.Second)
+			req, err = http.NewRequest("GET", "http://localhost:8091/pools/default", nil)
+			fatalError(`Failed to poll node status`, err)
+			req.SetBasicAuth("Administrator", "password")
+			response, err = client.Do(req)
+			fatalError(`Failed to poll node status`, err)
+			err = json.NewDecoder(response.Body).Decode(&jsonBody2)
+			fatalError(`Failed to poll node status`, err)
+		}
+
+		//Run integration tests
+		code := t.Run()
+
+		//Remove Docker container when complete
+		if err := pool.Purge(resource); err != nil {
+			log.Fatalf("Could not purge resource: %s", err)
+		}
+
+		os.Exit(code)
+	} else {
+		t.Run()
+	}
+}
+
+func areBucketsReady(res *[]responseBody) bool {
+	allReady := true
+	nres := *res
+	for i := 0; i < len(nres); i++ {
+		if len(nres[i].Nodes) == 1 && nres[i].Nodes[0].Status != "healthy" {
+			allReady = false
```

Comment:
Can `return false` here to stop the loop checking other nodes in the list when you know it's not ready.